src/widget.ts

```diff
@@ -15,13 +17,93 @@ import {
 // import '../css/widget.css'
 
 
+abstract class _ToneWidgetModel extends WidgetModel {
+
+  defaults () {
+    return {...super.defaults(),
+      _model_module: _ToneWidgetModel.model_module,
+      _model_module_version: _ToneWidgetModel.model_module_version,
+    };
+  }
+
+  static model_module = MODULE_NAME;
+  static model_module_version = MODULE_VERSION;
+
+}
+
+
+abstract class AudioNodeModel extends _ToneWidgetModel {
+
+  defaults() {
+    return {...super.defaults(),
+      _model_name: AudioNodeModel.model_name,
+      inputs: [],
+      outputs: [],
+    };
+  }
+
+  initialize (attributes: any, options: any) {
+    super.initialize(attributes, options);
+
+    this.node = this.createNode();
+
+    // TODO: remove this
+    this.node.toDestination();
+
+    this.initEventListeners();
+  }
+
+  initEventListeners() : void {
+  }
+
+  node: ToneAudioNode;
+
+  abstract createNode() : ToneAudioNode;
+
+  static model_name = 'AudioNodeModel';
+}
+
+
 export
-class OscillatorModel extends WidgetModel {
+class DestinationModel extends AudioNodeModel {
+  defaults() {
+    return {...super.defaults(),
+      _model_name: DestinationModel.model_name,
+      mute: false,
+      volume: -16
+    };
+  }
+
+  createNode () {
+    return getDestination();
```

Comment:
```suggestion
    return Tone.getDestination();
```

Reply:
I changed the way Tone is imported, but not sure it's a better way.